### PR TITLE
[DOCS-8513] Hide HTTP Client docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

Hide docs. Eng request.
- Removes from the nav
- Removes from the use case index pages
- Adds aliases to the HTTP doc in the use case index pages.
- Adds private to frontmatter.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/observability_pipelines/split_logs/http_client.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 title: Split Logs for HTTP Client
 disable_toc: false
+aliases:
```
Comment (drichards-87) [inline content/en/observability_pipelines/split_logs/http_client.md:4]: I think you might've accidentally added this alias (it's a circular redirect that points to itself/the same file).

**Review by drichards-87 — APPROVED** [review]
Left a note about a duplicate alias and approved the PR.

File: content/en/observability_pipelines/split_logs/http_client.md (hunk 1 above)
Comment (maycmlee) [reply content/en/observability_pipelines/split_logs/http_client.md:4]: Yes, you're right, thanks for catching that! I'll remove

File: content/en/observability_pipelines/split_logs/http_client.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+    - /observability_pipelines/split_logs/http_client/
```
Comment (maycmlee) [inline content/en/observability_pipelines/split_logs/http_client.md:5]:
```suggestion
```

**Comment by maycmlee** [comment]
Closing this out in favor of this update: https://github.com/DataDog/documentation/pull/24309

**Pull request closed without merging**